jetty-server/src/main/java/org/eclipse/jetty/server/ResourceService.java

```diff
@@ -832,7 +832,7 @@ protected void putHeaders(HttpServletResponse response, HttpContent content, lon
             if (_acceptRanges)
                 f.put(ACCEPT_RANGES);
 
-            if (_cacheControl != null)
+            if (_cacheControl != null && !response.containsHeader(HttpHeader.CACHE_CONTROL.asString()))
```

Comment:
We should do exactly the same for _acceptRanges above.

Reply:
done